### ISSUE TITLE
Configuration caching failure in Android projects

### ISSUE THREAD

**Issue by ILikeYourHat** [mikepenz/AboutLibraries#839]
## About this issue

I've found an issue with caching in `AboutLibrary` plugin, a pretty serious I think.

## Details

The recipe for disaster:
- create an Android project
- add this plugin to it, without any additional configuration (note that `registerAndroidTasks` is set to `true` in this case)
- create at least one additional Android module, and make it a `debugImplementation` dependency in the main module
- make sure to include some third-party dependencies in both modules (like AppCompat, nothing fancy)
- add those lines to the`gradle.properties` file: `org.gradle.configureondemand=true`, `org.gradle.unsafe.configuration-cache=true`
- try to execute `./gradlew :app:assembleRelease`

Expected outcome:
- a successful build

Actual outcome:
```
FAILURE: Build failed with an exception.

* What went wrong:
An exception occurred applying plugin request [id: 'com.android.library']
> Failed to apply plugin 'com.android.internal.library'.
   > Gradle#projectsEvaluated(Action) on build 'AboutLibrariesTestApp' cannot be executed in the current context.
```

So, in other words: assembling release application is broken.

I've prepared a sample application with this issue, reduced to the bare minimum needed to reproduce it:
https://github.com/ILikeYourHat/AboutLibrariesTestApp

## Checklist

- [x] Searched for [similar issues](https://github.com/mikepenz/AboutLibraries/issues)
- [x] Checked out the [sample application](https://github.com/mikepenz/AboutLibraries/tree/develop/app)
- [x] Read the [README](https://github.com/mikepenz/AboutLibraries/blob/develop/README.md)
- [x] Checked out the [CHANGELOG](https://github.com/mikepenz/AboutLibraries/releases)
- [ ] Read the [MIGRATION GUIDE](https://github.com/mikepenz/AboutLibraries/blob/develop/MIGRATION.md)

**Comment by mikepenz** [comment]
Thank you very much for the report @ILikeYourHat

**Comment by mikepenz** [comment]
I feel pretty positive that we can finally close this one after many years. v12.0.0-b03 and newer look very positive in my own use with config cache enabled.
